src/viur/core/modules/file.py

```diff
@@ -610,15 +660,20 @@ def write(self, filename: str, content: t.Any, mimetype: str = "text/plain", wid
         :param mimetype: The file's mimetype.
         :param width: Optional width information for the file.
         :param height: Optional height information for the file.
-
+        :param public: True if the file should be publicly accessible.
         :return: Returns the key of the file object written. This can be associated e.g. with a FileBone.
         """
         if not File.is_valid_filename(filename):
             raise ValueError(f"{filename=} is invalid")
 
         dl_key = utils.string.random()
 
-        blob = GOOGLE_STORAGE_BUCKET.blob(f"{dl_key}/source/{filename}")
+        if public:
+            dl_key += PUBLIC_DLKEY_SUFFIX  # mark file as public
+
+        bucket = File.get_bucket(dl_key)
```

Comment:
Please use `self` instead of the class to allow subclassing.
```suggestion
        bucket = self.get_bucket(dl_key)
```

(Please change all `File.$any` to `self.$any`)

And I would change external `File.$any` to `conf.main_app.file.$any`

Reply:
I would like to put this into a separate issue. There are also calls in render/html on this.

Reply:
Referenced by #1252, we shall keep it as is here for now.

Reply:
okay